src/math/bigintops.c

```diff
@@ -1161,16 +1184,8 @@ MVMObject * MVM_bigint_rand(MVMThreadContext *tc, MVMObject *type, MVMObject *b)
 
     if (use_small_arithmetic) {
         if (MP_GEN_RANDOM_MAX >= abs(smallint_max)) {
-            mp_digit result_int;
-#ifdef MP_NEW_LTM_VERSION
-            mp_digit p = MP_GEN_RANDOM_MAX;
-            mp_rand_digit(&p);
-            result_int = p;
-#else
-            result_int = MP_GEN_RANDOM();
-#endif
-            result_int = result_int % smallint_max;
-            if(have_to_negate)
+            mp_digit result_int = rand() % smallint_max;
```

Comment:
I'm pretty sure you shouldn't use `rand()` here

Reply:
I definitely wasn't 100% sure about this change (and it was a long enough time ago that I don't remember all that I tried), but `MP_GEN_RANDOM()` seems to be gone from libtommath v1.2.0 and `mp_rand_digit` is deprecated in favor of `mp_rand`, but I just want a random digit, not a whole `mp_int`. What do you suggest instead of `rand()`?

Reply:
can't you use `MVM_getrandom()`?

Reply:
Doh! Don't know why I didn't do that before. Just pushed a change to do so.